sql/src/main/java/io/crate/operation/projectors/ShardingUpsertExecutor.java

```diff
@@ -283,8 +283,13 @@ private void sendRequestsForBulk(CompletableFuture<BitSet> bulkResultFuture, Map
                 @Override
                 public void onResponse(ShardResponse shardResponse) {
                     nodeJobsCounter.decrement(shardLocation.nodeId);
-                    processShardResponse(shardResponse);
-                    countdown();
+                    Exception failure = shardResponse.failure();
+                    if (failure != null && failure instanceof InterruptedException) {
+                        bulkResultFuture.completeExceptionally(failure);
```

Comment:
I'm not sure if this here is good - it's the same behaviour as before. This causes the client to get an early "Job killed" error, but it's kind of fake as other requests may still be in progress.